### ISSUE TITLE
Outlook for the Web: Adding multiple attachments at the same time slows performance

### ISSUE THREAD

**Issue by JackTheDane** [OfficeDev/office-js#2095]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

Hi team!

We're encountering an issue which is affecting our performance. The issue appears to be that trying to add multiple inline attachments at the same time drastically increases the execution time, proportional to the amount of inline attachments, despite running each instance of `addFileAttachmentFromBase64Async` parallel using `Promises`.

## Expected Behavior
When running multiple `addFileAttachmentFromBase64Async` in parallel, execution time should not increase drastically compared to adding a single attachment.

## Current Behavior
In our tests, the more inline attachments that are added at the same time, the slower the script will execute.

Example of recorded time taken:
| Number of inline attachments  | Average script execution time |
| ------------- | ------------- |
| 1 attachment | 214ms  |
| 5 attachments | 1012ms |
| 11 attachments  | 2127ms  |

## Example GIST (Script lab for Outlook)
I have created a small demonstration of the problem, similar to our usage of the API, which can be found here: https://gist.github.com/JackTheDane/a781b51bfdb9d1979635b9f76e1a44d5.

In the example, you can add a variable number of attachments, and see how execution time increases for each attachment.

## Context
In our use-case, we're trying to insert email signatures, using inline attachments for images (due to the character limit of 30,000 characters on `setSignatureAsync`). We have noticed that signatures with many images take significantly longer time to complete than those with few or none, which affects the user experience.

## Your Environment

* Platform: Office Online
* Host: Outlook
* Operating System: Windows
* Browser: Google Chrome

## Useful logs
Example of waterfall/sequential behaviour, as seen in the Chrome network tab:
![image](https://user-images.githubusercontent.com/35668734/131668501-ffb9dcd5-c84f-4cc2-83a7-e715b7c69253.png)

Thank you for your time 🙏

**Comment by exextoc** [comment]
Thank you for raising the bug. We have put it in the backlog. We do not have any timeline to share as of now.
